src/Hosting/Hosting/test/HostingApplicationTests.cs

```diff
@@ -3,19 +3,90 @@
 
 using System.Collections;
 using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Hosting.Fakes;
 using Microsoft.AspNetCore.Hosting.Server.Abstractions;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.Testing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Metrics;
 using Moq;
 using static Microsoft.AspNetCore.Hosting.HostingApplication;
 
 namespace Microsoft.AspNetCore.Hosting.Tests;
 
 public class HostingApplicationTests
 {
+    [Fact]
+    public void Metrics()
+    {
+        // Arrange
+        var measurements = new Dictionary<string, long>();
+        void OnMeasurementRecorded(Instrument instrument, long measurement, ReadOnlySpan<KeyValuePair<string, object>> tags, object state)
+        {
+            measurements[instrument.Name] = measurement;
+            Console.WriteLine($"{instrument.Name} recorded measurement {measurement}");
+        }
+
+        using var meterListener = new MeterListener();
```

Comment:
What happens if the meter listener is created after the meter and its counters are created? Specifically, what happens to `InstrumentPublished`? Does it execute for the already existing counters?